chore(ci): update pr template (#1805)
ethereum/execution-spec-tests: .github/PULL_REQUEST_TEMPLATE.md

## 🗒️ Description
<!-- Brief description of the changes introduced by this PR -->

## 🔗 Related Issues or PRs
<!-- Reference any related issues using the GitHub issue number (e.g., Fixes #123). Default is N/A. -->
N/A.

## ✅ Checklist
<!-- Please check off all required items. For those that don't apply remove them accordingly. -->

- [ ] All: Set appropriate labels for the changes.

> - [ ] All: PR title and commit messages adhere to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles).

Reviewer: ```suggestion
- [ ] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope): `.
```

> - [ ] All: Considered squashing commits to improve commit history.

Reviewer: This is only relevant to merging PRs and only a maintainer can merge. And, perhaps most relevant, our merge policy now enforces squash and merge as a merge policy.
```suggestion
```

## 🗒️ Description
<!-- Brief description of the changes introduced by this PR -->
<!-- Don't submit this PR if it could expose a mainnet bug, see [SECURITY.md](/ethereum/execution-spec-tests/SECURITY.md) -->
<!-- Consider running the following fast checks before pushing changes to avoid unnecessary CI fails, see also "[Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/)"-->
<!-- uvx --with=tox-uv tox -e lint,typecheck,spellcheck,markdownlint -->

## 🔗 Related Issues or PRs
<!-- Reference any related issues using the GitHub issue number (e.g., Fixes #123). Default is N/A. -->
N/A.

## ✅ Checklist
<!-- Please check off all required items. For those that don't apply remove them accordingly. -->

> - [ ] All: Set appropriate labels for the changes (only maintainers can apply labels).

Reviewer: ```suggestion
- [ ] All: Ran fast `tox` checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    uvx --with=tox-uv tox -e lint,typecheck,spellcheck,markdownlint
    ```
```

Reviewer: This comment will often come too late: When a Contributor sees this page, they'll already have pushed their changes. But still, if they get a CI fail, they will immediately see here how they can remedy fails in fast checks and get a link to docs for other fails.

- [ ] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope): `.
- [ ] All: Considered adding an entry to [CHANGELOG.md](/ethereum/execution-spec-tests/blob/main/docs/CHANGELOG.md).
- [ ] All: Considered updating the online docs in the [./docs/](/ethereum/execution-spec-tests/blob/main/docs/) directory.

> - [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://eest.ethereum.org/main/tests/) are correctly formatted.

Reviewer: Move labels lower down in the Checklist  as Contributors can't apply labels.
```suggestion
- [ ] All: Set appropriate labels for the changes (only maintainers can apply labels).
- [ ] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://eest.ethereum.org/main/tests/) are correctly formatted.
```
